src/predict.rs

```diff
@@ -668,18 +670,20 @@ pub(crate) mod rust {
       v + acc
     });
     let avg = T::cast_from((sum + (width >> 1) as u32) / width as u32);
-    for line in output.rows_iter_mut().take(height) {
-      line[..width].iter_mut().for_each(|v| *v = avg);
+    for y in 0..height {
```

Comment:
Do not do that, we should assemble the border tiles/regions so they are aware they have some extra room while the other tiles/regions do not.

Reply:
Btw, I believe you aware this rust function does not run if SIMD available, which does what I have changed. I.e, cross the region border following av1 spec.

Reply:
So, I was not able to figure out other quick feasible way to achieve what I needed.

Reply:
> Do not do that, we should assemble the border tiles/regions so they are aware they have some extra room while the other tiles/regions do not.

Could you think of or suggest some other better ways to solve this? I've thought that we can split the function to have two versions, one for non frame boundary, and another for boundary case. But this split would be required every function affected by this PR and seemed not a good solution to me.

Reply:
> Do not do that, we should assemble the border tiles/regions so they are aware they have some extra room while the other tiles/regions do not.

Oh, I now understand. by 'border' you meant frame boundary. Then, this basically fall into the solution that I think and wanted to avoid.

Reply:
@lu-zero, I think the concern you have raised (that is valid) is cleared now, given that the src/predict.rs is intact now.
Can you +1 then?